### DISCUSSION TITLE
Why does `canvas.NewImageFromFile()` not load the image from the file system?

### DISCUSSION THREAD

**Discussion by srackham** [fyne-io/fyne#4435] [Q&A] [answered]
It just sets the file name:

https://github.com/fyne-io/fyne/blob/dc34dad76137def4466ab8eb4166e7ae741ef9ee/canvas/image.go#L195-L198

**Comment by andydotxyz** [comment, marked as answer]
Setting up an image just makes sure that the the data is available for the renderer to read the data (like FromReader or FromResource just sets up a resource in memory).

It's not until later that the image is decoded and rendered.

**Reply by srackham** [reply]
Thanks for clearing that up, I was playing with manually scaling and loading in-memory thumbnails (p.103 of your excellent _Building Cross-Platform GUI Applications with Fyne_ book) and I had completely forgotten that URI-based images are rendered auto-magically.
